luigi/contrib/hive.py

```diff
@@ -248,12 +272,60 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.transport.close()
 
 
+class WarehouseHiveClient(HiveClient):
+    """
+    Client for managed tables that makes decision based on presence of directory in hdfs
+    """
+
+    def __init__(self, hdfs_client=None, warehouse_location=None):
+        self.hdfs_client = hdfs_client or get_autoconfig_client()
+        self.warehouse_location = warehouse_location or get_hive_warehouse_location()
+
+    def table_schema(self, table, database='default'):
+        return NotImplemented
+
+    def table_location(self, table, database='default', partition=None):
+        return os.path.join(
+            self.warehouse_location,
+            database + '.db',
+            table,
+            self.partition_spec(partition)
+        )
+
+    def table_exists(self, table, database='default', partition=None):
+        """
+        We consider table/partition as existing if corresponding path in hdfs exists
```

Comment:
Can we remove the "we" wording?

`The table/partition is considered existing if corresponding path in hdfs exists and ....`